# Import the data

In [1]:
import warnings
warnings.filterwarnings('ignore')

In [2]:
#import data
import pandas as pd
import numpy as np
df = pd.read_csv('LoanStats3c_2014.csv',skiprows=1)

In [3]:
#delete column id and member_id
del df['id']
del df['member_id']

There are 235,631 data point int the raw data set and 135 features 

In [4]:
df.shape

(235631, 135)

# Extract first ten variables

In [5]:
#slice first ten varaibles in the df
df_10v = df[lambda df: df.columns[0:10]]
#check types
df_10v.dtypes

#clean the varaibles
df_10v['emp_length'] = df_10v['emp_length'].astype(str).str.replace('\D+', '')
df_10v['term'] = df_10v['term'].astype(str).str.replace('\D+', '')
df_10v['int_rate'] = df_10v['int_rate'].astype(str).str.replace('\D+', '')

df_10v['int_rate'] = pd.to_numeric(df_10v['int_rate'], errors='coerce')
df_10v['int_rate'] = df_10v['int_rate']/100

df_10v['term'] = pd.to_numeric(df_10v['term'], errors='coerce')
df_10v['emp_length'] = pd.to_numeric(df_10v['emp_length'], errors='coerce')

#rename intereste rate 
df_10v.rename(columns={'int_rate':'int_rate(%)'}, inplace=True)

In [6]:
#take a look the first ten variables
df_10v.head()

loan_amnt  funded_amnt  funded_amnt_inv  term  int_rate(%)  installment  \
0    10400.0      10400.0          10400.0  36.0         6.99       321.08   
1    15000.0      15000.0          15000.0  60.0        12.39       336.64   
2     7650.0       7650.0           7650.0  36.0        13.66       260.20   
3     9600.0       9600.0           9600.0  36.0        13.66       326.53   
4    12800.0      12800.0          12800.0  60.0        17.14       319.08   

  grade sub_grade                       emp_title  emp_length  
0     A        A3  Truck Driver Delivery Personel         8.0  
1     C        C1                      MANAGEMENT        10.0  
2     C        C3            Technical Specialist         1.0  
3     C        C3                Admin Specialist        10.0  
4     D        D4       Senior Sales Professional        10.0

# Numeric Variables in first 10 features

In [7]:
pd.set_option('precision', 2)
df_10v.describe(include=[np.number])

loan_amnt  funded_amnt  funded_amnt_inv       term  int_rate(%)  \
count  235629.00    235629.00        235629.00  235629.00    235629.00   
mean    14870.16     14870.16         14865.33      43.44        13.77   
std      8438.32      8438.32          8435.52      11.10         4.33   
min      1000.00      1000.00           950.00      36.00         6.00   
25%      8325.00      8325.00          8325.00      36.00        10.99   
50%     13000.00     13000.00         13000.00      36.00        13.65   
75%     20000.00     20000.00         20000.00      60.00        16.29   
max     35000.00     35000.00         35000.00      60.00        26.06   

       installment  emp_length  
count    235629.00   223610.00  
mean        442.46        6.19  
std         245.04        3.53  
min          23.36        1.00  
25%         265.68        3.00  
50%         384.06        7.00  
75%         578.69       10.00  
max        1409.99       10.00

# Categorical Variables in first 10 features

In [8]:
df_10v.describe(include=[np.object])

grade sub_grade emp_title
count   235629    235629    222393
unique       7        35     75352
top          C        C2   Teacher
freq     66565     13965      4569

# Extract 10th to 20th variables

In [18]:
df_20v = df[lambda df: df.columns[10:20]]
df_20v.count()

home_ownership         235629
annual_inc             235629
verification_status    235629
issue_d                235629
loan_status            235629
pymnt_plan             235629
url                         0
desc                    15279
purpose                235629
title                  235629
dtype: int64

Since the feature "desc" is the Loan description provided by the borrower, which varies from loaners to loaners. In addition this feature only have 15279 out of 235629, means more than 90% loaners did not provide this information. I decided to drop this variable. Same logic apply to feature url, which does not have any predictive value. I dropped it too.

In [19]:
df_20v = df_20v.drop('desc', 1)
df_20v = df_20v.drop('url', 1)
df_20v.head()

home_ownership  annual_inc verification_status   issue_d  loan_status  \
0       MORTGAGE     58000.0        Not Verified  Dec-2014  Charged Off   
1           RENT     78000.0     Source Verified  Dec-2014   Fully Paid   
2           RENT     50000.0     Source Verified  Dec-2014  Charged Off   
3           RENT     69000.0     Source Verified  Dec-2014   Fully Paid   
4       MORTGAGE    125000.0            Verified  Dec-2014      Current   

  pymnt_plan             purpose                    title  
0          n         credit_card  Credit card refinancing  
1          n  debt_consolidation       Debt consolidation  
2          n  debt_consolidation       Debt consolidation  
3          n  debt_consolidation       Debt consolidation  
4          n                 car            Car financing

# Numeric Variables 10th to 20th features¶

In [20]:
df_20v.describe(include=[np.number])

annual_inc
count    2.36e+05
mean     7.49e+04
std      5.55e+04
min      3.00e+03
25%      4.54e+04
50%      6.50e+04
75%      9.00e+04
max      7.50e+06

# Categorical Variables in 10-20th features

In [21]:
df_20v.describe(include=[np.object])

home_ownership verification_status   issue_d loan_status pymnt_plan  \
count          235629              235629    235629      235629     235629   
unique              4                   3        12           7          2   
top          MORTGAGE     Source Verified  Oct-2014  Fully Paid          n   
freq           119937               97741     38783      141525     235560   

                   purpose               title  
count               235629              235629  
unique                  13                2057  
top     debt_consolidation  Debt consolidation  
freq                143006              140624